# መግቢያ

ስለ ስታትስቲክስ ምንነት እና ብያኔ የስታትስቲክስ መጽሐፍ ምንበብ ይቻላል። በአጭሩ ግን ስታትስቲክስ "በአግባቡ" ጥቅም ላይ ሲውል፤ ውሂብ (ዳታ) ላይ የተመሰረተ እውቀት ለመሰብሰብ እና ውሳኔ ለማሳለፍ ይረዳል። በአግባቡ የሚለውን በጥቅስ ውስጥ ያስቀመጥኩት የትኛው የስታትስቲክስ መለኪያ ለየትኛው ትንተና አግባብ እንደሆነ መለየት እና መጠቀምን ስለሚፈልግ ነው። 

በዚህ ክፍል መሰረታዊ የሆኑትን ወይም በስታትስቲክስ ቋንቋ ገላጭ ስታትስቲክስ (ዲስክሪፕቲቭ ስታትስቲክስ) የምንላቸውን እንመለከታለን። የማዕከላዊነት መለኪያዎች (ሜዠር ኦፍ ሴንተራል ቴንደንሲ) እና የስብጥርነት መለኪያዎች (ሜዤር ኦፍ ስፕሪድ) የምንላቸውን እንመለከታለን። 

ለዚህም ይረዳን ዘንድ ፓይተን ፕሮግራም መጻፊያን ቋንቋ እንጠቀማለን። በተጨማሪም ሃሳቦችን በተግባር ለማየት ይረዳን ዘንድ አዲስ አበባ ዩኒቨርስቲ በ 2013 ካሳተመው የዩኒቨርስቲው መገለጫ (ፕሮፋይል) ላይ ያገኘሁትን ዊሂብ ወይም ዳታ እንጠቀማለን። ውሂብ በሚለው አቃፊ ውስጥ ታገኙታላችሁ። ውሂቡ በአአዩ እስከ 2010 ድረስ የተመረቁ ተማሪዎችን ብዛት በ መርሀግብር ከፋፍሎ የሚያሳይ ነው። 

*** 
መሰረታዊ የፓይተን እና የስታትስቲክስ እውቀት አላችሁ ተብሎ ይገመታል። 
*** 

# ውሂቡን ማንበብ

ውሂቡን ለማንበብ የውሂቡን አይነት ማወቅ አለብን። ይህንንም ከፋይሉ ቅጥያ መረዳት እንችላለን። CSV ፋይል ነው። እንደዚህ አይነት ፋይሎችን ለማንበብ የሚረዳ ፓይተን ውስጥ CSV የሚባል መገልገያ አለ። እሱን በመጥራት በቀላሉ ፋይሉን ማንበብ እንችላለን። 

In [1]:
# ውሂቡን በተለያየ መንገድ ማንበብ እንችላለን። 
# ለትንተናው ቀላል እንዲሆንልን በ ዝርዝር ላይ እናንበበው 
# csvን እንጥራው 
import csv
# ፋይሉን እናንበበው 
with open("./ውሂብ/የምረቃ_ውሂብ.csv", 'r', encoding='utf-8') as ፋይል:
    ውሂብ = list(csv.reader(ፋይል))

In [2]:
# ውሂቡ ምን እንደሚመስል ግንዛቤ ለመውሰድ የመጀመሪያ አምቱን ረድፍ እንመልከት። 
ውሂብ[:5]

[['የተመረቁበት ዓም', 'በ ሰርተፍኬት ', 'በዲፕሎማ', 'በዲግሪ', 'በድሕረ ምረቃ'],
 ['1952', '', '11', '', ''],
 ['1953', '', '9', '', ''],
 ['1954', '', '5', '13', ''],
 ['1955', '11', '44', '9', '']]

# ገላጭ ስታትስቲክስ

ውሂብ (ዳታ) ልናጠና ስለፈለግነው ክስተት ወይም ድርጊት ምልከታችንን መዝግበን የምንይዝበት ነው። ለምሳሌ የተጠቀምነው ውሂብ በአአዩ ከ 1952 እስክ 2010 ድረስ በዩኒቨርስቲው የተመረቁ ተማሪዎችን መዝግቦ የያዘ ሰነድ ነው። ውሂቡን በመጠቀም ትንተናዎችን ማካሄድ እንችላለን። ለምሳሌ በአማካይ ዩኒቨርስቲው በየዓመቱ ምን ያህል ተማሪዎችን ያስመርቃል? ከፍተኛው እና ዝቅተኛው የተማሪ ቁጥር ምረቃ የተካሄደው መቼ ነው? እነዚህ እና ተዛማጅ የሆኑ ጥያቄዎችን መመለስ እንችላለን። 

## የማዕከላዊነት መለኪያዎች  

### አማካኝ (ሚን)

አማካኝ ($\bar{አ}$) ወይም ሚን የምንለው የማዕከላዊነት መለኪያ ነው። አማካይ እንደሚከተለው ይቀመራል። 

*** 
$\bar{አ} = \frac{\sum{አ}}{ብ} \Rightarrow $ አ (የእያንዳንዱ ምልከታ እሴት)፤ ብ(የእሴቶቹ ቆጠራ ብዛት) 
*** 

ይኼ ማለት በውሂባችን ከሚገኙ ምልከታዎች አንዱን እሴት በዘፈቀደ ብንወስድ ከአማካዩ ጋር የተቀራረበ (አማካዩን እራሱ ካልሆነ) እሴት እናገኛለን ማለት ነው።   

In [15]:
# የዲፕሎማ ተማሪዎችን እንውሰድ እና አማካዩን እንስራ
# የዲፕሎ ተመራቂ ተማሪዎች ሦስተኛውን አምድ (ኮለመን) እንወስዳለን። 
# በፓይተን ሦስተኛው አምድ በ 2 ይወከላል። የአምድ ጠቋሚ ቁጥር ከ ዜሮ ስለሚጀምር። 
ዲፕሎማ_ተመራቂ = [float(ዲ[2]) for ዲ in ውሂብ[1:]] 

In [16]:
ዲፕሎማ_ተመራቂ[:10]

[11.0, 9.0, 5.0, 44.0, 1.0, 1.0, 31.0, 49.0, 28.0, 18.0]

In [17]:
# አጠቃላይ የዲፕሎማ ተመራቂዎች ድምር 
አጠቃላይ_ዲፕሎማ_ተመራቂ = sum(ዲፕሎማ_ተመራቂ)

In [18]:
#አጠቃላይ ምልከታ ብዛት 
የምልከታ_ብዛት=len(ዲፕሎማ_ተመራቂ)

In [19]:
# አማካኙ እንደሚከተለው ይሆናል ማለት ነው። 
አማካኝ = አጠቃላይ_ዲፕሎማ_ተመራቂ/የምልከታ_ብዛት

In [20]:
አማካኝ

770.7796610169491

ይኼ እንደሚነግረን አዲስ አበባ ዩኒቨርስቲ በአማካኝ በየዓመቱ 771 ተማሪዎችን በዲፕሎማ ያስመርቃል ማለት ነው። አማካኙን ላይ ብቻ ተንተርሰን ውሳኔ ላይ መድረስ አንችልም። ከ አማካኝ ጋር ዝምድና ያለው የማእከላዊነት መለኪያ አጋማሽ (ሚዲያን) የምንለው ነው።  

### አጋማሽ (ሚዲያን)

አጋማሽ ማለት መካከለኛው እሴት ማለት ነው። መካከለኛው እሴት ስንል፣ ውሂባችንን  በረድፍ ከላይ ወደታች ካስያዝን ቦኋላ መካከለኛውን እሴት መውሰድ ማለት ነው። የውሂቡ እርዝማኔ ሙሉ ቁጥር ከሆነ የመካከለኛ ሁለቱን አማካኝ እንደ አጋማሽ እንወስደዋለን። 

In [21]:
# የዲፕሎማ ተመራቂዎችን አጋማሽ ፍለጋ 
ዲፕሎማ_ተመራቂ = [float(ዲ[2]) for ዲ in ውሂብ[1:]] 
#አጠቃላይ ምልከታ ብዛት 
የምልከታ_ብዛት=len(ዲፕሎማ_ተመራቂ)


In [23]:
ዲፕሎማ_ተመራቂ[:10]

[11.0, 9.0, 5.0, 44.0, 1.0, 1.0, 31.0, 49.0, 28.0, 18.0]

In [24]:
የምልከታ_ብዛት

59

In [26]:
#የምልከታ ብዛቱ ጎዶሎ ቁጥር ስለሆነ 
# የአጋማሹ ጠቋሚ (የምልከታ_ብዛት/2)+0.5 ይሆናል 
(የምልከታ_ብዛት/2)+0.5

30.0

In [27]:
# ሰላሳኛውን እሴት ለማግኘት ውሂባችንን ከትንሽ ወደትልቅ ረድፍ እናስይዝ 
ረድፍ_የያዘ_ዲ_ተመራቂ=sorted(ዲፕሎማ_ተመራቂ)

In [28]:
ረድፍ_የያዘ_ዲ_ተመራቂ[:10]

[1.0, 1.0, 5.0, 9.0, 11.0, 15.0, 18.0, 24.0, 26.0, 28.0]

In [34]:
# ሰላሳኛው እሴት ማለት በ 29ነኛው ረድፍ የሚገኘው ነው (ፓይተን ጠቋሚ መቁጠር የሚጀምረው ከ ዜሮ ነው።)
#እሱንም ለማግኘት ረድፍ_የያዘ_ዲ_ተመራቂ[29] ማሮጥ ነው። 
ረድፍ_የያዘ_ዲ_ተመራቂ[29]

679.0

አጋማሹ 679 ሲሆን አማካኙ 771 ነው። እንደምንመለከተው ሁለቱ ልዩነት አል። አማካኝ ለጽንፍ ወይም አውትላየር የተጋለጠ ነው። አጋማሹ በጽንፍ አይጠቃም። አማካኙ ሁሉንም እሴቶች ከግንዛቤ አስገብቶ ስለሚደምር፣ በጽንፍ የሚገኙ እሴቶች ተጽኖ ያሳድሩበታል። እስቲ ትንሹን እና ትልቁን እሴት በውሂባችን እንመልከት።  

In [35]:
# ትልቁን እና ትንሹን እሴት ከውሂባችን ውስጥ እንፈልግ። 
ዲፕሎማ_ተመራቂ = [float(ዲ[2]) for ዲ in ውሂብ[1:]]
አነስተኛ, ከፍተኛ = [min(ዲፕሎማ_ተመራቂ), max(ዲፕሎማ_ተመራቂ)]

In [36]:
አነስተኛ, ከፍተኛ

(1.0, 3299.0)

ጽንፍ በውሂባችን እንደሚገኝ ማረጋገጥ እንችላለን። አጋማሽ 679 እና አማካኝ 771 በሆነበት 1 ተመራቂ መገኘት ወይም 3299 ተመረቂ መገኘት ጽንፍ መኖሩን ያረጋግጥልናል። ጽንፎቹም ውሂቡን በምንሰበስብበት ግዜ የተፈጠረ ስህተትን ጠቋሚ ሊሆኑ ይችላሉ። ላይሆኑም ይችላሉ። ነገር ግን በውሂባችን ውስጥ መኖራቸውን ማስተዋሉ ጥሩ ነው። 

### ተደጋጋሚው እሴት (ሞድ)

ሌላኛው የማእከላዊነት መለኪያው ተደጋጋሚው እሴት ወይም ሞድ የምንለው ነው። ተደጋጋሚው እሴት (በውሂቡ ውስጥ ከመደጋገሙ የተነሳ) በአማካኙ ላይ ተጽኖ ይኖረዋል። ተደጋጋሚው እሴት እንደሚከተለው ማግኘት እንችላለን።  

In [38]:
#ተደጋጋሚውን እሴት (ሞድ) ለማግኘት 
# ባዶ መዝገበቃል እናውጅ
ተደጋጋሚ_ቆጠራ = {}
ዲፕሎማ_ተመራቂ = [float(ዲ[2]) for ዲ in ውሂብ[1:]] 
for እሴት in ዲፕሎማ_ተመራቂ:
    if እሴት not in ተደጋጋሚ_ቆጠራ:
        ተደጋጋሚ_ቆጠራ[እሴት] =1
    else:
        ተደጋጋሚ_ቆጠራ[እሴት]+=1
        
# ተደጋጋሚውን እሴት ለማግኘት 
# ተደጋጋሚው ዜሮ ግዜ ነው የተደጋገመው ብለን ታሳቢ አድርገን እንነሳ
የተደጋጋሚው_ቆጠራ_ብዛት = 0
ተደጋጋሚው_እሴት = None

for ቁልፍ, እሴት in ተደጋጋሚ_ቆጠራ.items():
    if የተደጋጋሚው_ቆጠራ_ብዛት < እሴት:
        የተደጋጋሚው_ቆጠራ_ብዛት = እሴት
        ተደጋጋሚው_እሴት=ቁልፍ
print(ተደጋጋሚው_እሴት, የተደጋጋሚው_ቆጠራ_ብዛት)

1.0 2


ከላይ እንደተመለከትነው ተደጋጋሚው እሴት 1 ሲሆን የተደጋገመውም 2 ጊዜ ነው። ስለዚህ የማእከላዊነት መለኪያችንን ብንመለከት አማካኙ (771)፣ አጋማሹ (679) እና ተደጋጋሚው እሴት 1 እሱም በውሂባችን ውስጥ አነስተኛው እሴት ነው።  የተራራቁ መሆኖቸው ስለውሂባችን የሚሰጠን ፍንጭ ቁጥሮቹ በአስተማማኝነት የዲፕሎማ መርሀግብርን ውሂብ ይወክላሉ ለማለት አያስደፍርም። 

# ማጠቃለያ

በዚህ ክፍል የማዕከላዊነት መለኪያ የሆኑትን ስታትስቲካዊ ትንተና መስጫዎችን ፓይተንንን በመጠቀም እንዴት መስራት እንደምንችል ተመልክተናል። በሚቀጥለው ክፍል የስብጥር መለኪያ የሆኑትን እንመለከታለን።  